.github/workflows/pr-merged.yml

```diff
@@ -0,0 +1,177 @@
+name: PR Merged
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions:
+  contents: read
+
+jobs:
+  merged:
+    name: Show info
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-24.04-arm
+    steps:
+      # github.event.pull_request.base may not refer to the _actual_ parent commit of the PR,
+      # instead it is the commit that the PR was merged/rebased onto.
+      #
+      # Since we want to diff the PR's commits, query the first commit's first parent.
+      #
+      # Alternatively, we could compare
+      # github.event.pull_request.merge_commit_sha to github.event.pull_request.base.sha,
+      # however I'm unsure how that'll interact with grouped Merge Queue 🤔
+      - name: Get base commit
+        id: base
+        env:
+          GH_TOKEN: ${{ github.token }}
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+          pr: ${{ github.event.pull_request.number }}
+        run: |
+          out=$(mktemp)
+          gh api graphql -F owner="$owner" -F repo="$repo" -F pr="$pr" -f query='
+              query($owner: String!, $repo: String!, $pr: Int!) {
+                repository(owner: $owner, name: $repo) {
+                  pullRequest(number: $pr) {
+                    commits(first: 1) {
+                      edges {
+                        node {
+                          commit {
+                            parents(first: 1) {
+                              edges {
+                                node {
+                                  oid
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+              ' --jq '
+                .data
+                .repository
+                .pullRequest
+                .commits
+                .edges[0]
+                .node
+                .commit
+                .parents
+                .edges[0]
+                .node
+                .oid
+              ' > "$out"
+          echo "sha=$(cat "$out")" >> "$GITHUB_OUTPUT"
+
+      # Checkout CI scripts from the latest commit from the base branch.
+      # This ensures we get the latest scripts, even if the PR's base was outdated.
+      - name: Checkout base branch
+        id: checkout
+        uses: actions/checkout@v4
+        # TODO: sparse checkout
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+        with:
+          extra_nix_config: |
+            accept-flake-config = true
+
+      # NOTE: This is here for debugging and gathering data
+      # TODO: Remove
```

Comment:
What is the time scope for this `TODO: Remove`?

Reply:
Once we understand how PRs merged in grouped batches (by the queue) show up.

Likely it will be removed the next time we touch this workflow, hopefully to integrate it with a "generate message" script.